src/services/create-service.test.js

```diff
@@ -0,0 +1,107 @@
+import { createSpace } from './create-service.js';
+import test from 'tape';
+
+test('create-service', t => {
+
+  t.test('createSpace', t => {
+    t.test('throws NOT YET IMPLEMENTED on multiple components', t => {
+      createSpace(null, '', '', [1, 2])
+      .catch(err => {
+        t.true(err.message.includes('NOT YET IMPLEMENTED'));
+        t.end();
+      });
+    });
+
+
+    t.test('throws on schemas that don\'t meet Spaces API', t => {
+      const space = 'my-clay-space',
+        store = {
+          state: {
+            schemas: {
+              [space]: {
+                content: {
+                  _componentList: {
+                    include: ['somthing else']
+                  }
+                }
+              }
+            }
+          }
+        };
+
+      createSpace(store, '', '', [space])
+      .catch(err => {
+        // un-comment to see what the error message is
+        // t.comment(err.message);
+        t.true(err.message.includes('Check the schema.yml'));
+        t.end();
+        return true;
+      });
+    });
+
+    t.test('instantiates logic, embeds component', t => {
+      const dispatchCalls = [],
+        space = 'my-clay-space',
+        spaceLogic = 'my-space-logic',
+        path = 'my-path',
+        ref = 'ref',
+        parentRef = 'parentRef',
+        availableSpaces = [space],
+        store = {
```

Comment:
A test like this makes me smell something smelly.
Maybe I should be passing less stuff into createSpace or at least handling the "extract the stuff I want from the store" stuff further up. 

Suggestions would be much appreciated. Once consideration is that we get the best chance of correct behavior if property access on the store isn't cached (the store state, such as `ui.currentSelection` can change over the course of space-creation and logic embedding).